packages/expo-modules-autolinking/scripts/ios/autolinking_manager.rb

```diff
@@ -42,15 +42,23 @@ class AutolinkingManager
             end
 
             podspec_dir_path = Pathname.new(pod.podspec_dir).relative_path_from(project_directory).to_path
+            podspec_file_path = File.join(podspec_dir_path, pod.pod_name + ".podspec")
+            podspec = Pod::Specification.from_file(podspec_file_path)
+
+            # Ensure that the dependencies of packages with Swift code use modular headers, otherwise
+            # `pod install` may fail if there is no `use_modular_headers!` declaration or
+            # `:modular_headers => true` is not used for this particular dependency.
+            # The latter require adding transitive dependencies to user's Podfile that we'd rather like to avoid.
+            if package.has_swift_modules_to_link?
+              use_modular_headers_for_dependencies(podspec.all_dependencies)
+            end
```

Comment:
```suggestion
            # Ensure that the dependencies of packages with Swift code use modular headers, otherwise
            # `pod install` may fail if there is no `use_modular_headers!` declaration or
            # `:modular_headers => true` is not used for this particular dependency.
            # The latter require adding transitive dependencies to user's Podfile that we'd rather like to avoid.
            if package.has_swift_modules_to_link?
              podspec_file_path = File.join(podspec_dir_path, pod.pod_name + ".podspec")
              podspec = Pod::Specification.from_file(podspec_file_path)
              use_modular_headers_for_dependencies(podspec.all_dependencies)
            end
```

Reply:
moving the podspec parsing into the if block, so that we don't have to parse the spec if the package has no swift code.